src/duplicacy_snapshotmanager.go

```diff
@@ -597,8 +598,8 @@ func (manager *SnapshotManager) ListAllFiles(storage Storage, top string) (allFi
 	return allFiles, allSizes
 }
 
-// GetSnapshotChunks returns all chunks referenced by a given snapshot.
-func (manager *SnapshotManager) GetSnapshotChunks(snapshot *Snapshot) (chunks []string) {
+// CacheSnapshotChunks returns all chunks referenced by a given snapshot.
+func (manager *SnapshotManager) CacheSnapshotChunks(snapshot *Snapshot) (chunks []string) {
```

Comment:
I think instead of keeping two versions of GetSnapshotChunks and LoadChunks, it is possible to add an argument to the original GetSnapshotChunks that indicates if `snapshot.ChunkHashes` needs to be cleared at the end of the call, like:

```
func (manager *SnapshotManager) GetSnapshotChunks(snapshot *Snapshot, clearChunkHashes bool) (chunks []string) {
```

The only place where `clearChunkHashes` should be false is inside `ShowStatisticsTabular` where `GetSnapshotChunks` is called twice on all snapshots.  In all other places I believe it is safe to set `clearChunkHashes` to true.

Reply:
I considered both approaches (parameter vs separate methods).
- The logic for `GetSnapshotChunks` using a parameter is not too complicated, I'd probably name the parameter something like `cacheChunks` and reverse the logic though, since the zero value for bool is `false`, and the default behaviour sounds like it should be not to cache the chunks in memory.
- The problem for me is with usage of `LoadChunks`, since this is used directly in a few places.  If this is retained in its original state, storing data on the snapshot object, it becomes the responsibility of every calling site to `nil` that field if they don't want to cache the chunks, which is just asking for unexpected memory usage.  `LoadChunks` also doesn't really lend itself well to operating on a parameter for caching, since it needs to return the chunks for the non-caching version, but also store them on the snapshot when caching.

If you would prefer not to retain two implementations, one option would be to replace the caching variant of `LoadChunks` with the implementation in this PR, which will return the chunks without storing them, then store them on the snapshot inside `GetSnapshotChunks`, based on a parameter.  The reason I didn't consider this route is because I wasn't sure how `LoadChunks` was used in places like `DownloadSnapshotSequence`, and I didn't have time to work back up the call chain(s) for these other methods.

Reply:
What I thought was to keep `LoadChunks` as it is.  Other than `GetSnapshotChunks`, it is only called by `DownloadSnapshotSequence`, which in turn is called by `DownloadSnapshotContents`.  It should be okay to keep `snapshot.ChunkHashes` in ``DownloadSnapshotContents` I think.

Feel free to reverse the logic for `clearChunkHashes` and give it a different name; just avoid using `cache` in its name.  This would create some confusion with the snapshot cache.